streams/src/main/java/org/apache/kafka/streams/kstream/ValueTransformerSupplier.java

```diff
@@ -18,9 +18,14 @@
 package org.apache.kafka.streams.kstream;
 
 /**
- * A value transformer supplier which can create one or more {@link ValueTransformer} instances.
+ * A {@link ValueTransformerSupplier} interface which can create one or more {@link ValueTransformer} instances.
```

Comment:
Always new instance on each invocation. This is because multiple instances may need to have its own local variables and we cannot share a single instance across multiple tasks / threads.